### PR TITLE
Added the policy_template context method.

### PR DESCRIPTION
**Pull request by Superhepper**
Signed-off-by: Jesper Brynolf <jesper.brynolf@gmail.com>

### CODE REVIEW

File: tss-esapi/src/context/tpm_commands/enhanced_authorization_ea_commands.rs [hunk 1]
```diff
@@ -454,6 +454,71 @@ impl Context {
...
+                self.optional_session_3(),
+                &template_hash.clone().into(),
+            )
+        };
+        let ret = Error::from_tss_rc(ret);
+        if ret.is_success() {
+            Ok(())
+        } else {
+            error!(
+                "Failed to bind template has to a specific creation template: {}",
```
Comment (ionut-arm) [inline tss-esapi/src/context/tpm_commands/enhanced_authorization_ea_commands.rs:517]:
```suggestion
                "Failed to bind template to a specific creation template: {}",
```
?

File: tss-esapi/src/context/tpm_commands/enhanced_authorization_ea_commands.rs [hunk 2]
```diff
@@ -454,6 +454,71 @@ impl Context {
...
+    /// #     )
+    /// #     .expect("Failed to create session")
+    /// #     .expect("Recived invalid handle");
+    /// # let template_hash = Digest::try_from(vec![
+    /// #     1, 2, 3, 4, 5, 6, 7, 8, 9, 10,
+    /// #     11, 12, 13, 14, 15, 16, 17, 18, 19, 20
+    /// # ]).expect("Failed to create template hash");
+    /// // Result should be checked because the TPM might not support this.
+    /// let _ = context.policy_template(trial_session, &template_hash)
+    ///     .expect("Call to policy_template failed");
```
Comment (ionut-arm) [inline tss-esapi/src/context/tpm_commands/enhanced_authorization_ea_commands.rs:495]: Hmm, is this example adding much to the documentation? I think if the documentation for the parameters is clear enough (which it seems to be!) and the setup doesn't require complicated steps then we can just skip the example

Comment (Superhepper) [reply tss-esapi/src/context/tpm_commands/enhanced_authorization_ea_commands.rs:495]: Nah I think you are right. The only interesting in this example is the comment. But that should be added to the detailed section instead.

**Review by puiterwijk — APPROVED** [review]
LGTM

**Review by ionut-arm — APPROVED** [review]
Thank you!

**Pull request merged**